File: examples/python/basic.py
"""
Query String Builder API - Basic Usage Example

This example demonstrates the basic usage of the Query String Builder API.
API Documentation: https://docs.apiverve.com/ref/querystringbuilder
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/querystringbuilder'

def call_querystringbuilder_api():
    """
    Make a POST request to the Query String Builder API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;params&#x27;: {
        &#x27;name&#x27;: &#x27;John Doe&#x27;,
        &#x27;age&#x27;: 30,
        &#x27;city&#x27;: &#x27;New York&#x27;,
        &#x27;interests&#x27;: [
            &#x27;coding&#x27;,
            &#x27;music&#x27;,
            &#x27;travel&#x27;
        ]
    },
    &#x27;encode&#x27;: true
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Query String Builder API...\n')

    result = call_querystringbuilder_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Query String Builder API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Query String Builder API.
 * API Documentation: https://docs.apiverve.com/ref/querystringbuilder
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/querystringbuilder';

/**
 * Make a POST request to the Query String Builder API
 */
async function callQueryStringBuilderAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;params&quot;: {
        &quot;name&quot;: &quot;John Doe&quot;,
        &quot;age&quot;: 30,
        &quot;city&quot;: &quot;New York&quot;,
        &quot;interests&quot;: [
            &quot;coding&quot;,
            &quot;music&quot;,
            &quot;travel&quot;
        ]
    },
    &quot;encode&quot;: true
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callQueryStringBuilderAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
